### PR TITLE
Stop pushing telemetry reports from the proxy

### PR DESCRIPTION
**Pull request by olix0r**
Now that the controller does not depend on pushed telemetry reports, the
proxy need not depend on the telemetry API or maintain legacy sampling
logic.

Depends on #615

### CODE REVIEW

**Comment by olix0r** [comment]
This is blocked on the Query endpoint being ported to use the new prometheus data (#499)

**Comment by briansmith** [comment]
> +23 −780
> 
> \o/

\o/. I love the thinking behind this how series of changes in how we communicate metrics to Prometheus!

I am going to defer to the other reviewers since I'm not so familiar with this code and there's no point getting familiar with it during its removal!

File: proxy/src/telemetry/control.rs [hunk 1]
```diff
@@ -38,11 +32,8 @@ pub struct MakeControl {
 /// # TODO
 /// Limit the amount of memory that may be consumed for metrics aggregation.
 pub struct Control {
-    /// Holds the current state of aggregated metrics.
-    push_metrics: Option<PushMetrics>,
-
     /// Aggregates scrapable metrics.
-    metrics_work: prometheus::Aggregate,
+    metrics_aggregate: prometheus::Aggregate,
```
Comment (hawkw) [inline proxy/src/telemetry/control.rs]: 👍

File: proxy/src/telemetry/control.rs [hunk 2]
```diff
@@ -197,82 +132,37 @@ impl Control {
...
-                    self.reset_timeout();
-                    Ok(Async::Ready(Some(report)))
-                }
-            }
         }
     }
 }
 
-// NOTE: `flush_timeout` does not impl `Debug`.
 impl fmt::Debug for Control {
```
Comment (hawkw) [inline proxy/src/telemetry/control.rs]: Can this impl be derived now that the flush timeout has been removed?

**Comment by hawkw** [comment]
It looks like #499 is done, so we can move forwards with this again.

**Comment by olix0r** [comment]
This will merge following https://github.com/runconduit/conduit/pull/753

**Comment by hawkw** [comment]
Now that this has merged, I think it might be worthwhile to reorganize the proxy's `telemetry` module somewhat (but not necessarily in this release).

**Pull request merged**